docs/roadmap.md

```diff
@@ -0,0 +1,268 @@
+
+  - NVIDIA Device Plugin
+
+  - DCGM Exporter
+
+  - vGPU manager
+
+#### Con: Driver installer by default downloads and compiles at runtime
+
+The default configuration runs a container image that downloads and
+installs OS packages and then downloads, compiles and installs the
+NVIDIA driver kernel modules - this is all done by the DaemonSet's Pod
+when it starts on the GPU node. We used to do something similar for AI
+Core, but found the approach to be somewhat fragile as well as adding a
+significant amount of time to the node startup phase.
+
+It is possible to tell the operator to use "precompiled" images instead,
+which results in a similar approach to how AI Core is installing the
+NVIDIA driver. Of course, a build pipeline must be set up to create
+these images.
+
+Both types of image (download & compile; precompiled) are built from the
+[NVIDIA GPU Driver
+Container](https://github.com/NVIDIA/gpu-driver-container) repo. The
+root of this repo contains folders for various operating systems.
+
+Only Ubuntu 22.04 and Ubuntu 24.04 are officially supported for
+precompiled images, although the repo also contains the required files
+and instructions to build precompiled images for RHEL 8 and RHEL 9.
+
+#### Con: Garden Linux is not a supported platform
+
+The NVIDIA GPU Operator supports only Ubuntu and Red Hat operating
+systems. In principle, support for Garden Linux could be added
+reasonably easily - however NVIDIA might not accept PRs for Garden Linux
+support and therefore we might need to use and maintain a fork.
+
+#### Con: NVIDIA Container Runtime requires host OS configuration
+
+See [Installing the NVIDIA Container Toolkit — NVIDIA Container
+Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html#configuring-containerd-for-kubernetes)
+
+The NVIDIA Container Toolkit requires a functioning package manager on
+the host OS, but the Garden Linux read-only filesystem prevents new
+packages from being installed. This is probably the biggest barrier to
+getting things working.
+
+## Roadmap for the future
+
+### Step 1 - Add Garden Linux support to the NVIDIA GPU Operator
+
+There are several sub-steps here:
+
+1.  Add support for Garden Linux in the [NVIDIA GPU Driver
+    Container](https://github.com/NVIDIA/gpu-driver-container) repo
+
+    It should be possible to use the Ubuntu examples from the [NVIDIA GPU
+Driver Container](https://github.com/NVIDIA/gpu-driver-container) repo
+in combination with the existing Garden Linux scripts to synthesise
+Garden Linux support for both default and precompiled images.
+
+
+2.  Figure out how to install the NVIDIA Container Toolkit/Runtime on
+    Garden Linux
+
+    The toolkit itself is [open-source on
+GitHub](https://github.com/NVIDIA/nvidia-container-toolkit) so we
+might be able to figure out an alternative way to install it. In the
+worst case we would need to build a specific Garden Linux image to
+support NVIDIA GPUs.
```

Comment:
Ah, so pre-installed images then. Is that driver-version-independent? What kind of compatibility matrix/issues are to be expected?

Reply:
Regarding your [first comment](https://github.com/gardenlinux/gardenlinux-nvidia-installer/pull/42#discussion_r2090692090): 
The configuration aspect is addressed in [another comment above](https://github.com/gardenlinux/gardenlinux-nvidia-installer/pull/42#discussion_r2091402228).

Regarding your [second comment](https://github.com/gardenlinux/gardenlinux-nvidia-installer/pull/42#discussion_r2090870641):
The specific Garden Linux images we are talking about here contain only the NVIDIA Container Runtime files (**not** the kernel modules for the driver) - these are Go binaries stored in `/usr/bin` and `/usr/lib` and are independent of the kernel and driver versions, so there is no compatibility matrix to consider (other than keeping the binaries more-or-less up to date).